mmeds/tools/qiime2.py

```diff
@@ -554,25 +557,49 @@ def composition_ancom(self, category, level=None):
         ]
         self.jobtext.append(' '.join(cmd))
 
-    def taxa_collapse(self, category, taxa_level):
-        """ Collapse taxonomy to the specified level """
-        new_file = '{}_table_l{}'.format(category, taxa_level)
-        self.add_path(new_file, '.qza')
+    def collapse_all_taxa_relative(self, taxa_levels=(1, 6)):
+        """ Collapse taxonomy at the specified levels, merge, and convert to relative abundance """
+        self.taxa_collapse(taxa_levels)
+        self.merge_collapsed_taxa(taxa_levels)
+        self.taxa_absolute_to_relative_abundance()
+
+    def taxa_collapse(self, taxa_levels):
+        """ Collapse taxonomy at specified taxa levels (inclusive) """
+        for taxa_level in range(taxa_levels[0], taxa_levels[1]+1):
+            new_file = 'taxa_collapsed_table_L{}'.format(taxa_level)
+            self.add_path(new_file, '.qza')
+            cmd = [
+                'qiime taxa collapse',
+                '--i-table {}'.format(self.get_file('filtered_table')),
+                '--i-taxonomy {}'.format(self.get_file('taxonomy')),
+                '--p-level {}'.format(taxa_level),
+                '--o-collapsed-table {}'.format(self.get_file(new_file))
+            ]
+            self.jobtext.append(' '.join(cmd))
+
+    def merge_collapsed_taxa(self, taxa_levels):
+        """ Merge collapsed taxa files to make one complete collapsed file as in qiime1's summarize_taxa.py """
+        collapsed_tables = []
+        for taxa_level in range(taxa_levels[0], taxa_levels[1]+1):
+            collapsed_tables.append(str(self.get_file(f'taxa_collapsed_table_L{taxa_level}')))
```

Comment:
is this a dataframe append?

Reply:
nope, just a list, so a bunch of tables can get merged in the next command